---
title: C28205
ms.date: 11/04/2016
ms.topic: reference
f1_keywords:
- C28205
helpviewer_keywords:
- C28205
ms.assetid: 3d802885-bdb8-47cb-9108-527a328a0774
author: mikeblome
ms.author: mblome
manager: markl
ms.workload:
- multiple
ms.openlocfilehash: 237bbc33acde5b84ae549184ef8743c478f988f1
ms.sourcegitcommit: 8e123bcb21279f2770b28696995450270b4ec0e9
ms.translationtype: MT
ms.contentlocale: pt-BR
ms.lasthandoff: 12/25/2019
ms.locfileid: "75400707"
---
# <a name="c28205"></a>C28205
aviso C28205: função >: `_Success_` ou `_On_failure_` usado em um contexto inválido

 As anotações `_Success_` e `_On_failure_` só podem ser usadas em valores de retorno de função.

## <a name="examples"></a>Exemplos

```cpp
#include<sal.h>

// Oops, _Success_ is not valid in parameter lists, should be moved to return value.
bool GetValue( _Success_(return != false) _Out_ int *pInt, bool flag)
{
   if(flag) {
      *pInt = 5;
      return true;
   } else {
      return false;
   }
}
```

```cpp
#include<sal.h>

_Success_(return != false)
bool GetValue(_Out_ int *pInt, bool flag)
{
   if(flag) {
      *pInt = 5;
      return true;
   } else {
      return false;
   }
}
```

[Compreendendo sal](understanding-sal.md)
